Building with eloquent from source fails

**Describe the bug**

Building `pendulum_demo` and `pendulum_manager` along with the source repos (on `master` fails).

```
--- stderr: pendulum_controller_node                                                                                                              
CMake Error at CMakeLists.txt:24 (find_package):
  By not providing "Findrclcpp_components.cmake" in CMAKE_MODULE_PATH this
  project has asked CMake to find a package configuration file provided by
  "rclcpp_components", but CMake did not find one.

  Could not find a package configuration file provided by "rclcpp_components"
  with any of the following names:

    rclcpp_componentsConfig.cmake
    rclcpp_components-config.cmake

  Add the installation prefix of "rclcpp_components" to CMAKE_PREFIX_PATH or
  set "rclcpp_components_DIR" to a directory containing one of the above
  files.  If "rclcpp_components" provides a separate development package or
  SDK, be sure it has been installed.


---
Failed   <<< pendulum_controller_node   [ Exited with code 1 ]
--- stderr: pendulum_manager
CMake Error at CMakeLists.txt:21 (find_package):
  By not providing "Findrclcpp_components.cmake" in CMAKE_MODULE_PATH this
  project has asked CMake to find a package configuration file provided by
  "rclcpp_components", but CMake did not find one.

  Could not find a package configuration file provided by "rclcpp_components"
  with any of the following names:

    rclcpp_componentsConfig.cmake
    rclcpp_components-config.cmake

  Add the installation prefix of "rclcpp_components" to CMAKE_PREFIX_PATH or
  set "rclcpp_components_DIR" to a directory containing one of the above
  files.  If "rclcpp_components" provides a separate development package or
  SDK, be sure it has been installed.


---
Aborted  <<< pendulum_manager
--- stderr: pendulum_driver                                             
CMake Error at CMakeLists.txt:24 (find_package):
  By not providing "Findrclcpp_components.cmake" in CMAKE_MODULE_PATH this
  project has asked CMake to find a package configuration file provided by
  "rclcpp_components", but CMake did not find one.

  Could not find a package configuration file provided by "rclcpp_components"
  with any of the following names:

    rclcpp_componentsConfig.cmake
    rclcpp_components-config.cmake

  Add the installation prefix of "rclcpp_components" to CMAKE_PREFIX_PATH or
  set "rclcpp_components_DIR" to a directory containing one of the above
  files.  If "rclcpp_components" provides a separate development package or
  SDK, be sure it has been installed.


---
Aborted  <<< pendulum_driver

Summary: 123 packages finished [7min 2s]
  1 package failed: pendulum_controller_node
  2 packages aborted: pendulum_driver pendulum_manager
  10 packages had stderr output: cyclonedds foonathan_memory_vendor pendulum_controller_node pendulum_driver pendulum_manager rmw_connext_cpp rmw_connext_shared_cpp rosidl_typesupport_connext_c rosidl_typesupport_connext_cpp tracetools
  3 packages not processed
```

Seems like some packages do not declare their dependency on `rclcpp_components`. If I add it to their `package.xml` file, I now get this:

```
--- stderr: pendulum_simulation                                                                                  
In file included from /home/chris/ros2_ws/src/pendulum/pendulum_driver/pendulum_simulation/include/pendulum_simulation/pendulum_simulation.hpp:27:0,
                 from /home/chris/ros2_ws/src/pendulum/pendulum_driver/pendulum_simulation/src/pendulum_simulation.cpp:15:
/home/chris/ros2_ws/install/pendulum_driver/include/pendulum_driver/pendulum_driver_interface.hpp:21:10: fatal error: sensor_msgs/msg/joint_state.hpp: No such file or directory
 #include "sensor_msgs/msg/joint_state.hpp"
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/pendulum_simulation.dir/src/pendulum_simulation.cpp.o] Error 1
make[1]: *** [CMakeFiles/pendulum_simulation.dir/all] Error 2
make: *** [all] Error 2
---
Failed   <<< pendulum_simulation        [ Exited with code 2 ]
Aborted  <<< pendulum_controllers                                        

Summary: 130 packages finished [53.4s]
  1 package failed: pendulum_simulation
  1 package aborted: pendulum_controllers
  1 package had stderr output: pendulum_simulation
  1 package not processed
```

**To Reproduce**

* Setup a workspace with the `master` source repos.
* Add this repo (using the default branch, which seems to be `eloquent`).
* Build, e.g. `colcon build --packages-up-to pendulum_demo pendulum_manager`

**Expected behavior**

It builds.

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04

**Additional context**

Note that I'm not using the `eloquent` source, but it should probably still work. `sensor_msgs` hasn't changed.

--- a/pendulum_controller/pendulum_controller_node/CMakeLists.txt
+++ b/pendulum_controller/pendulum_controller_node/CMakeLists.txt
@@ -17,6 +17,7 @@ endif()
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
 find_package(ament_cmake REQUIRED)
+find_package(sensor_msgs REQUIRED)
 find_package(pendulum_msgs_v2 REQUIRED)
 find_package(rclcpp_lifecycle REQUIRED)
 find_package(lifecycle_msgs REQUIRED)
@@ -59,6 +60,7 @@ ament_target_dependencies(pendulum_controller_node
         "rclcpp"
         "rclcpp_components"
         "pendulum_tools"
+        "sensor_msgs"
         ${OPTIONAL_AMENT_DEPENDENCES})
 rclcpp_components_register_nodes(pendulum_controller_node "pendulum::Controller")
 target_link_libraries(pendulum_controller_node ${OPTIONAL_LIBRARIES})

--- a/pendulum_controller/pendulum_controller_node/package.xml
+++ b/pendulum_controller/pendulum_controller_node/package.xml
@@ -9,14 +9,18 @@
 
   <buildtool_depend>ament_cmake</buildtool_depend>
 
+  <build_depend>sensor_msgs</build_depend>
   <build_depend>pendulum_msgs_v2</build_depend>
   <build_depend>rclcpp_lifecycle</build_depend>
+  <build_depend>rclcpp_components</build_depend>
   <build_depend>lifecycle_msgs</build_depend>
   <build_depend>osrf_testing_tools_cpp</build_depend>
   <build_depend>pendulum_tools</build_depend>
 
+  <exec_depend>sensor_msgs</exec_depend>
   <exec_depend>pendulum_msgs_v2</exec_depend>
   <exec_depend>rclcpp_lifecycle</exec_depend>
+  <exec_depend>rclcpp_components</exec_depend>
   <exec_depend>lifecycle_msgs</exec_depend>
   <exec_depend>ros2run</exec_depend>
   <exec_depend>pendulum_tools</exec_depend>

--- a/pendulum_demo/package.xml
+++ b/pendulum_demo/package.xml
@@ -16,6 +16,7 @@
   <build_depend>pendulum_tools</build_depend>
   <build_depend>pendulum_msgs_v2</build_depend>
   <build_depend>rclcpp_lifecycle</build_depend>
+  <build_depend>rclcpp_components</build_depend>
   <build_depend>rcutils</build_depend>
   <build_depend>osrf_testing_tools_cpp</build_depend>
   <build_depend>tlsf_cpp</build_depend>
@@ -28,6 +29,8 @@
   <exec_depend>pendulum_controllers</exec_depend>
   <exec_depend>pendulum_controller_node</exec_depend>
   <exec_depend>lifecycle_msgs</exec_depend>
+  <exec_depend>rclcpp_lifecycle</exec_depend>
+  <exec_depend>rclcpp_components</exec_depend>
   <exec_depend>ros2run</exec_depend>
   <exec_depend>sensor_msgs</exec_depend>
 

--- a/pendulum_driver/pendulum_driver/package.xml
+++ b/pendulum_driver/pendulum_driver/package.xml
@@ -12,12 +12,14 @@
   <build_depend>lifecycle_msgs</build_depend>
   <build_depend>pendulum_msgs_v2</build_depend>
   <build_depend>rclcpp_lifecycle</build_depend>
+  <build_depend>rclcpp_components</build_depend>
   <build_depend>osrf_testing_tools_cpp</build_depend>
   <build_depend>pendulum_tools</build_depend>
   <build_depend>sensor_msgs</build_depend>
 
   <exec_depend>pendulum_msgs_v2</exec_depend>
   <exec_depend>rclcpp_lifecycle</exec_depend>
+  <exec_depend>rclcpp_components</exec_depend>
   <exec_depend>lifecycle_msgs</exec_depend>
   <exec_depend>ros2run</exec_depend>
   <exec_depend>pendulum_tools</exec_depend>

--- a/pendulum_driver/pendulum_simulation/CMakeLists.txt
+++ b/pendulum_driver/pendulum_simulation/CMakeLists.txt
@@ -12,6 +12,7 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
 endif()
 
 find_package(ament_cmake REQUIRED)
+find_package(sensor_msgs REQUIRED)
 find_package(pendulum_msgs_v2 REQUIRED)
 find_package(pendulum_driver REQUIRED)
 
@@ -32,6 +33,7 @@ ament_export_libraries(pendulum_simulation)
 
 ament_target_dependencies(pendulum_simulation
         "pendulum_driver"
+        "sensor_msgs"
         "pendulum_msgs_v2")
 
 # since the package installs libraries without exporting them

--- a/pendulum_driver/pendulum_simulation/package.xml
+++ b/pendulum_driver/pendulum_simulation/package.xml
@@ -9,13 +9,13 @@
 
   <buildtool_depend>ament_cmake</buildtool_depend>
 
+  <build_depend>sensor_msgs</build_depend>
   <build_depend>pendulum_msgs_v2</build_depend>
   <build_depend>pendulum_driver</build_depend>
-  <build_depend>sensor_msgs</build_depend>
 
+  <exec_depend>sensor_msgs</exec_depend>
   <exec_depend>pendulum_msgs_v2</exec_depend>
   <exec_depend>pendulum_driver</exec_depend>
-  <exec_depend>sensor_msgs</exec_depend>
 
   <test_depend>ament_cmake_gmock</test_depend>
   <test_depend>ament_cmake_gtest</test_depend>

--- a/pendulum_manager/CMakeLists.txt
+++ b/pendulum_manager/CMakeLists.txt
@@ -14,6 +14,7 @@ endif()
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
 find_package(ament_cmake REQUIRED)
+find_package(sensor_msgs REQUIRED)
 find_package(pendulum_msgs_v2 REQUIRED)
 find_package(rclcpp_lifecycle REQUIRED)
 find_package(lifecycle_msgs REQUIRED)

--- a/pendulum_manager/package.xml
+++ b/pendulum_manager/package.xml
@@ -9,12 +9,16 @@
 
   <buildtool_depend>ament_cmake</buildtool_depend>
 
+  <build_depend>sensor_msgs</build_depend>
   <build_depend>pendulum_msgs_v2</build_depend>
   <build_depend>rclcpp_lifecycle</build_depend>
+  <build_depend>rclcpp_components</build_depend>
   <build_depend>lifecycle_msgs</build_depend>
 
+  <exec_depend>sensor_msgs</exec_depend>
   <exec_depend>pendulum_msgs_v2</exec_depend>
   <exec_depend>rclcpp_lifecycle</exec_depend>
+  <exec_depend>rclcpp_components</exec_depend>
   <exec_depend>lifecycle_msgs</exec_depend>
   <exec_depend>ros2run</exec_depend>
 

